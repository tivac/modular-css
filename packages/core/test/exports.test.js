"use strict";

var dedent = require("dedent"),
    namer  = require("test-utils/namer.js"),
    
    Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("exports", () => {
        var processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer
            });
        });
        
        it("should export an object of arrays containing strings", () =>
            processor.string(
                "./simple.css",
                dedent(`
                    .red { color: red; }
                    .black { background: #000; }
                    .one, .two { composes: red, black; }
                `)
            )
            .then((result) =>
                expect(result.exports).toMatchSnapshot()
            )
        );

        it("should export identifiers and their classes", () =>
            processor.file(
                "./packages/core/test/specimens/start.css"
            )
            .then(() => processor.output())
            .then((output) =>
                expect(output.compositions).toMatchSnapshot()
            )
        );
    });
});
