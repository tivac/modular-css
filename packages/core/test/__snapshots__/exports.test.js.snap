// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js exports should export an object of arrays containing strings 1`] = `
Object {
  "black": Array [
    "black",
  ],
  "one": Array [
    "red",
    "black",
    "one",
  ],
  "red": Array [
    "red",
  ],
  "two": Array [
    "red",
    "black",
    "two",
  ],
}
`;

exports[`/processor.js exports should export identifiers and their classes 1`] = `
Object {
  "packages/core/test/specimens/folder/folder.css": Object {
    "folder": "folder",
  },
  "packages/core/test/specimens/local.css": Object {
    "booga": "booga",
    "folder": "white",
    "looga": "booga looga",
    "one": "red",
    "two": "blue",
  },
  "packages/core/test/specimens/start.css": Object {
    "booga": "booga",
    "folder": "white",
    "one": "red",
    "tooga": "tooga",
    "two": "blue",
    "wooga": "booga wooga",
  },
}
`;
